numhask/src/NumHask/Data/LogField.hs

```diff
@@ -0,0 +1,500 @@
+{-# LANGUAGE FlexibleInstances, FlexibleContexts, UndecidableInstances, MultiParamTypeClasses #-}
+module NumHask.Data.LogField where
+
+import           GHC.Generics                   ( Generic
+                                                , Generic1
+                                                )
+import           Data.Data                      ( Data )
+
+import           NumHask.Algebra.Additive
+import           NumHask.Algebra.Multiplicative
+import           NumHask.Algebra.Distribution
+import           NumHask.Algebra.Field
+import           NumHask.Algebra.Integral
+import           NumHask.Algebra.Rational
+
+import           Prelude                 hiding ( Num(..)
+                                                , negate
+                                                , sin
+                                                , cos
+                                                , sqrt
+                                                , (/)
+                                                , atan
+                                                , pi
+                                                , exp
+                                                , log
+                                                , recip
+                                                , (**)
+                                                , toInteger
+                                                )
+import qualified Data.Foldable                 as F
+
+-- LogField is adapted from LogFloat
+----------------------------------------------------------------
+--                                                  ~ 2015.08.06
+-- |
+-- Module      :  Data.Number.LogFloat
+-- Copyright   :  Copyright (c) 2007--2015 wren gayle romano
+-- License     :  BSD3
+-- Maintainer  :  wren@community.haskell.org
+-- Stability   :  stable
+-- Portability :  portable (with CPP, FFI)
+-- Link        :  https://hackage.haskell.org/package/logfloat
+----------------------------------------------------------------
+
+----------------------------------------------------------------
+--
+-- | A @LogField@ is just a 'Field' with a special interpretation.
+-- The 'LogField' function is presented instead of the constructor,
+-- in order to ensure semantic conversion. At present the 'Show'
+-- instance will convert back to the normal-domain, and hence will
+-- underflow at that point. This behavior may change in the future.
+--
+-- Because 'logField' performs the semantic conversion, we can use
+-- operators which say what we *mean* rather than saying what we're
+-- actually doing to the underlying representation. That is,
+-- equivalences like the following are true[1] thanks to type-class
+-- overloading:
+--
+-- > logField (p + q) == logField p + logField q
+-- > logField (p * q) == logField p * logField q
+--
+-- TODO: what to do with this note?
```

Comment:
I think we briefly summarise the motivation of the original library, credit it, and just say that this is the numhask version.